core/types/transaction.go

```diff
@@ -934,25 +870,26 @@ func (s TxByNonce) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 type TxWithMinerFee struct {
 	tx       *Transaction
 	minerFee *big.Int
+	received time.Time
 }
 
-func (tx *TxWithMinerFee) Tx() *Transaction   { return tx.tx }
-func (tx *TxWithMinerFee) MinerFee() *big.Int { return tx.minerFee }
+func (tx *TxWithMinerFee) Tx() *Transaction    { return tx.tx }
+func (tx *TxWithMinerFee) MinerFee() *big.Int  { return tx.minerFee }
+func (tx *TxWithMinerFee) Received() time.Time { return tx.received }
 
 // NewTxWithMinerFee creates a wrapped transaction, calculating the effective
 // miner gasTipCap if a base fee is provided.
 // Returns error in case of a negative effective miner gasTipCap.
-func NewTxWithMinerFee(tx *Transaction, baseFee *big.Int, qiTxFee *big.Int) (*TxWithMinerFee, error) {
+func NewTxWithMinerFee(tx *Transaction, qiTxFee *big.Int, received time.Time) (*TxWithMinerFee, error) {
 	if tx.Type() == QiTxType {
 		return &TxWithMinerFee{
 			tx:       tx,
 			minerFee: qiTxFee,
+			received: received,
 		}, nil
 	}
-	minerFee, err := tx.EffectiveGasTip(baseFee)
-	if err != nil {
-		return nil, err
-	}
+	// minerFee is now the baseFee and minerTip
+	minerFee := tx.GasPrice()
```

Comment:
Can you add a comment or make an issue that this needs to be updated? Because we are prioritizing based on GasPrice instead of the total fee

Reply:
I will make an issue

Reply:
https://github.com/dominant-strategies/go-quai/issues/2164